sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -256,15 +268,23 @@ public <T> T insert(String containerName, T objectToSave, PartitionKey partition
     @SuppressWarnings("unchecked")
     public <S extends T, T> Iterable<S> insertAll(CosmosEntityInformation<T, ?> information, Iterable<S> entities) {
         Assert.notNull(entities, "entities to be inserted should not be null");
-
         String containerName = information.getContainerName();
         Class<T> domainType = information.getJavaType();
-
         List<CosmosItemOperation> cosmosItemOperations = new ArrayList<>();
+        Map<String, HashMap<Field, Object>> mapOfTransientFieldValuesMaps = new HashMap<>();
```

Comment:
Because the method returns a Flux of bulk operations (multiple domain objects). So, storing id of each corresponding doc and a map of transient fields and values for that doc. This was originally a map storing just id and original object, now it is a map of maps to store the id, plus map of transient values.